src/main/java/com/pubnub/api/endpoints/History.java

```diff
@@ -182,23 +207,37 @@ private JsonElement processMessage(JsonElement message) throws PubNubException {
         String outputText;
         JsonElement outputObject;
 
-        if (mapper.isJsonObject(message) && mapper.hasField(message, "pn_other")) {
-            inputText = mapper.elementToString(message, "pn_other");
+        if (mapper.isJsonObject(message)) {
+            if (mapper.hasField(message, PN_OTHER)) {
+                inputText = mapper.elementToString(message, PN_OTHER);
+            } else {
+                PubNubError error = logAndReturnDecryptionError();
+                throw new PubNubException(error.getMessage(), error, null, null, 0, null, null);
+            }
         } else {
             inputText = mapper.elementToString(message);
         }
-
-        outputText = CryptoModuleKt.decryptString(cryptoModule, inputText);
+        try {
+            outputText = CryptoModuleKt.decryptString(cryptoModule, inputText);
+        } catch (com.pubnub.api.crypto.exception.PubNubException e) {
+            PubNubError error = logAndReturnDecryptionError();
+            throw new PubNubException(error.getMessage(), error, message, null, 0, null, null);
```

Comment:
this line looks different that for the same method in FetchMessages. Is this on purpose?

throw new PubNubException(error.getMessage(), error, null, null, 0, null, null);

Reply:
nope, it should be null. fixed